h2/src/main/org/h2/engine/Session.java

```diff
@@ -320,8 +319,8 @@ public void readSessionState() {
         sessionState = Utils.newSmallArrayList();
         CommandInterface ci = prepareCommand(!isOldInformationSchema()
                 ? "SELECT STATE_COMMAND FROM INFORMATION_SCHEMA.SESSION_STATE"
-                : "SELECT SQL FROM INFORMATION_SCHEMA.SESSION_STATE", Integer.MAX_VALUE);
-        ResultInterface result = ci.executeQuery(0, false);
+                : "SELECT SQL FROM INFORMATION_SCHEMA.SESSION_STATE");
+        ResultInterface result = ci.executeQuery(0, Integer.MAX_VALUE, false);
```

Comment:
why not -1 here?

Reply:
Because `ci` is actually a `CommandRemote` and it needs some sane value.

(But the whole method can be moved to `SessionRemote`.)